### DISCUSSION TITLE
Last station style used is used as next station style, rather than default station style.

### DISCUSSION THREAD

**Discussion by karst** [OpenRCT2/OpenRCT2#15898] [Feature requests]
### Operating System

Windows 10, 64 Bit

### OpenRCT2 build

OpenRCT2, v0.3.4.1 (0c9511c on refs/pull/10664/merge) provided by GitHub

### Describe the issue

When selecting a new ride, the ride will use the last selected station theming on another ride. Rather than going for the default RCT2 entrance with no theming (Plain style). 
To be completely honest, this might be a cool thing as a feature to have, but it feels really wacky and I would personally prefer to use it the old way. If this was an intended feature I would like to see an option where I can turn it off or a drop down menu where I can select what I want as default station theming. 

### Area(s) with issue?

New Save Format (NSF)

### Steps to reproduce

1. Place down a ride (for example Bumper Cars)
2. Set the style of your ride station to something different than Plain. (For example Abstract theme)
3. Place a new ride (for example a Wooden Rollercoaster)
4. As you see this ride will now also use Abstract theming. rather than Plain theming.

**Comment by seanmajorpayne** [comment]
@Gymnasiast I can work on this. Which menu should this show up in?

**Comment by Gymnasiast** [comment]
@seanmajorpayne This is the result of a deliberate (but not necessarily final) decision, so it’s not really a bug you can fix.

----------------------

The reason behind the change is that the code used to simply create all new rides with station style 0, which in SV6 is always the "plain" style. In the new save format, this is no longer guaranteed. In fact, index 0 is not even guaranteed to have any station style at all.

There are, as far as I can see, three ways to approach this:
- Use the last entrance style (like we do now)
- Loop through the entrance style and check of `rct2.station.plain` is in there and take that index (falling back to the first occupied index)
- Taking the first index that has a station style

**Comment by seanmajorpayne** [comment]
Ah I see. No worries. If you all decide you want to change it, let me know and I'll update. I'm indifferent to the behavior.

**Comment by toimine** [comment]
An another option could be being able to select the default station style somewhere in ride options

**Comment by ocalhoun6** [comment]
But then what happens when the user's default station style isn't available in the scenario they're currently playing?

**Comment by toimine** [comment]
You'd set it per scenario

**Comment by Derpercoaster** [comment]
I believe that, in the vanilla RCT1 game, this behavior was normal. Which ever station you last used on a ride was automatically chosen for your next ride.

**Comment by Gymnasiast** [comment]
@toimine 
> An another option could be being able to select the default station style somewhere in ride options

Please, no. We don’t want dozens of options for stuff that is too small to really care about. It clutters up both the Options window and the code.

@Derpercoaster 
> I believe that, in the vanilla RCT1 game, this behavior was normal. Which ever station you last used on a ride was automatically chosen for your next ride.

I just checked and it seems you’re right. However, this is not _necessarily_ a reason for OpenRCT2 to do it as well.


I am going to move this to Discussions, as it’s a matter of preference and not a bug.
